### PR TITLE
🚧(api) seed the experience index on bootstrap

### PR DESCRIPTION
**Pull request by lebaudantoine**
## Purpose

Populate frontend from the Experience Index.

## Proposal

Seed the experience index while bootstrapping the project with mocked data.

### CODE REVIEW

File: bin/seed_experience_index.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
+
+"""Activity statements randomization distribution"""
```
Comment (jmaupetit) [inline bin/seed_experience_index.py:2]: copy/pasta fail :sweat_smile:

File: bin/seed_experience_index.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
+
```
Comment (jmaupetit) [inline bin/seed_experience_index.py:1]:
```suggestion
```

File: bin/seed_experience_index.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    # Get a database session
+    session = next(get_session())
+
+    # Pass the database session to the factory
+    ExperienceFactory.__session__ = session
+
+    # Restore pristine state of the Experience Index
+    session.exec(delete(Relation))
+    session.exec(delete(Experience))
+    session.commit()
```
Comment (jmaupetit) [inline bin/seed_experience_index.py:31]: I think this should be handled separately. IMO this script's responsibility is to create/update XI-related records, not trashing existing data.

File: bin/seed_experience_index.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+            "title": '{"en":"Lesson 10"}',
+            "structure": "atomic",
+            "aggregation_level": 1,
+        },
+    ]
+
+    # Seed the Experience Index with mocked data
+    session.add_all(
+        ExperienceFactory.build(**r) for r in mocked_data
+    )
```
Comment (jmaupetit) [inline bin/seed_experience_index.py:100]: A more concise way would be to create `mocked_data` as a `(iri, title)` tuple list and hard code `structure` and `aggregation_level`:

```suggestion
    session.add_all(
        ExperienceFactory.build(
            iri=iri, 
            title= f'{{"en": "{title}"}}',
            structure="atomic", 
            aggregation_level=1
        ) for iri, title in mocked_data
    )
```

File: bin/seed_experience_index.py [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+    # Pass the database session to the factory
+    ExperienceFactory.__session__ = session
+
+    # Restore pristine state of the Experience Index
+    session.exec(delete(Relation))
+    session.exec(delete(Experience))
+    session.commit()
+
+    # Declare mocked data
+    mocked_data = [
```
[the remark this answers is not in the document]
Comment (jmaupetit) [reply bin/seed_experience_index.py:34]: JSON files are harder to maintain than code. WDYT?

File: bin/seed_experience_index.py (hunk 4 above)
Comment (jmaupetit) [reply bin/seed_experience_index.py:100]: Or even all at once:

```python
    session.add_all(
        ExperienceFactory.build(
            iri=iri, 
            title= f'{{"en": "lesson {i}"}}',
            structure="atomic", 
            aggregation_level=1
        ) for i, iri in enumerate(
            [
                "uuid://0aecfa93-cef3-45ae-b7f5-a603e9e45f50",
                "uuid://1c0c127a-f121-4bd1-8db6-918605c2645d",
                "uuid://541dab6b-50ae-4444-b230-494f0621f132",
                "uuid://69d32ad5-3af5-4160-a995-87e09da6865c",
                "uuid://7d4f3c70-1e79-4243-9b7d-166076ce8bfb",
                "uuid://8d386f48-3baa-4acf-8a46-0f2be4ae243e",
                "uuid://b172ec09-97ec-4651-bc57-6eabebf47ed0",
                "uuid://d613b564-5d18-4238-a69c-0fc8cee5d0e7",
                "uuid://dd38149d-956a-483d-8975-c1506de1e1a9",
                "uuid://e151ee65-7a72-478c-ac57-8a02f19e748b",
            ]
        )
    )
```

**Review by jmaupetit — COMMENTED** [review]
Good idea!

File: bin/seed_experience_index.py (hunk 5 above)
Comment (quitterie-lcs) [inline bin/seed_experience_index.py:34]: To get a more compact code, can you declare it with a `for` loop, iterate on lesson number and generate an IRI per item ?

[the remark this answers is not in the document]
Comment (lebaudantoine) [reply bin/seed_experience_index.py:34]: It might, trust you.

Comment (lebaudantoine) [reply bin/seed_experience_index.py:34]: My goal wasn't to have the more compact code ever. The IRI have to match the statement patched in the LRS.

File: bin/seed_experience_index.py (hunk 4 above)
Comment (lebaudantoine) [reply bin/seed_experience_index.py:100]: Yeah I thought about it, but at first I wanted to extract any data to a dedicated file. I'll for sure refactor it.

File: bin/seed_experience_index.py (hunk 3 above)
Comment (lebaudantoine) [reply bin/seed_experience_index.py:31]: Alright, so, I'll just seed the database, and ignore when errors happen if the IRI already exist.

File: bin/seed_experience_index.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+from warren.db import get_session
+from warren.xi.factories import ExperienceFactory
+from warren.xi.models import (
+    Experience,
+    Relation,
+)
+
+logger = logging.getLogger(__name__)
+
+def seed_experience_index():
```
Comment (jmaupetit) [inline bin/seed_experience_index.py:18]:
```suggestion
logger = logging.getLogger(__name__)


def seed_experience_index():
```

**Review by jmaupetit — APPROVED** [review]

**Review by quitterie-lcs — APPROVED** [review]

**Pull request merged**